pymediainfo/__init__.py

```diff
@@ -346,11 +351,23 @@ def parse(cls, filename, library_file=None, cover_data=False,
                 )
             for option_name, option_value in mediainfo_options.items():
                 lib.MediaInfo_Option(handle, option_name, option_value)
-        if lib.MediaInfo_Open(handle, filename) == 0:
-            lib.MediaInfo_Close(handle)
-            lib.MediaInfo_Delete(handle)
-            raise RuntimeError("An error occured while opening {}"
-                    " with libmediainfo".format(filename))
+
+        try:
+            file_bytes = filename.read()
+            opener = lib.MediaInfo_Open_Buffer
+        except AttributeError:
+            opener = lib.MediaInfo_Open
+
+        if opener is lib.MediaInfo_Open_Buffer:
+            if not isinstance(file_bytes, bytes):
+                raise ValueError("Given file should be opened in binary mode.")
+            res = lib.MediaInfo_Open_Buffer(handle, file_bytes, len(file_bytes), b"", 0)
```

Comment:
Yes and no, nice spot.

The first time I wrote it, I checked the res value (both res, after the if/else), but after reading the C++ code I spotted that it can never return 0, so I moved the check back, and forgot to remove the name binding.